src/raftstore/store/worker/read.rs

```diff
@@ -433,21 +452,14 @@ impl<'r, 'm> RequestInspector for Inspector<'r, 'm> {
 
     fn inspect_lease(&mut self) -> LeaseState {
         // TODO: disable localreader if we did not enable raft's check_quorum.
-        if let Some(ref lease) = self.delegate.leader_lease {
-            // None means now.
-            if LeaseState::Valid == lease.inspect(None) {
-                return LeaseState::Valid;
-            }
-            self.metrics.rejected_by_lease_expire += 1;
-            debug!(
-                "{} leader lease is expired: {:?}",
-                self.delegate.tag, self.delegate.leader_lease
-            );
+        if self.delegate.leader_lease.is_some() {
+            // We skip lease check, because it is postponed until `handle_read`.
+            LeaseState::Valid
```

Comment:
So it is possible for the function to return that a lease is valid when it is not?

Reply:
Yes, but we will check it later in `handle_read` and we will find if it is invalid.

Reply:
My concern is later we will use `inspect_lease` and not remember this gotcha. It could be dangerous. Maybe you can return a value that conveys this? Like `Option<LeaseState>`? Or add a `LeaseState::Postponed`?

Reply:
I like the idea `LeaseState::Postponed`. It is reflected in #3510.